### PR TITLE
fixed validation for bundle checkbox options

### LINKED ISSUE #27489: JS validate: show many messsages for Bundle product with checkbox option
### Preconditions (*)

1. Fresh Magento 2.3.4

### Steps to reproduce (*)

1. Create bundle product
2. Add option. Option Name: anything; Input Type: Checkbox; Required
3. Add product to Option
4. Add to cart on store front.
5. From validate if no option selected.
image

### Expected result (*)

1. JS should validate and show one error message for Option

### Actual result (*)

1. Show many messages
image

### PR DESCRIPTION
**Pull request by ProkopovVitaliy**
### Description (*)

JS should validate and show one error message for Option

### Fixed Issues (if relevant)

1. magento/magento2#27489: Issue title JS validate: show many messsages for Bundle product with checkbox option

### Manual testing scenarios (*)

1. Create bundle product
2. Add option. Option Name: anything; Input Type: Checkbox; Required
3. Add product to Option
4. Add to cart on store front.
5. From validate if no option selected.
### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by rogyar** [comment]
Hi @ProkopovVitaliy. Thank you for the fix, it's very useful. Would you cover your change with MFTF test, please? 

Thank you!

**Comment by ProkopovVitaliy** [comment]
@rogyar covered with MFTF test )

File: app/code/Magento/Bundle/Test/Mftf/ActionGroup/AssertStorefrontBundleValidationCountActionGroup.xml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<actionGroups xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+              xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/actionGroupSchema.xsd">
+    <actionGroup name="AssertStorefrontBundleValidationCountActionGroup">
```
Comment (rogyar) [inline app/code/Magento/Bundle/Test/Mftf/ActionGroup/AssertStorefrontBundleValidationCountActionGroup.xml]:
```suggestion
    <actionGroup name="AssertStorefrontBundleValidationMessagesCountActionGroup">
```

File: app/code/Magento/Bundle/Test/Mftf/ActionGroup/AssertStorefrontBundleValidationCountActionGroup.xml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        <annotations>
+            <description>Check if it exists validation message box on page and their number</description>
```
Comment (rogyar) [inline app/code/Magento/Bundle/Test/Mftf/ActionGroup/AssertStorefrontBundleValidationCountActionGroup.xml]:
```suggestion
            <description>Check if there's a validation message box on page and asserts the validation messages number</description>
```

File: app/code/Magento/Bundle/Test/Mftf/Test/StorefrontBundleCheckBoxOptionValidationTest.xml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+                <requiredEntity createDataKey="simpleProduct1"/>
+                <field key="qty">2</field>
+            </createData>
+            <createData entity="ApiBundleLink" stepKey="createBundleLink2">
+                <requiredEntity createDataKey="bundleProduct"/>
+                <requiredEntity createDataKey="checkboxBundleOption"/>
+                <requiredEntity createDataKey="simpleProduct2"/>
+                <field key="qty">4</field>
+            </createData>
+            <magentoCLI command="indexer:reindex" arguments="cataloginventory_stock" stepKey="reindex"/>
```
Comment (rogyar) [inline app/code/Magento/Bundle/Test/Mftf/Test/StorefrontBundleCheckBoxOptionValidationTest.xml]: Please, do not run the indexer directly, use the corresponding cron command

https://github.com/magento/magento-functional-tests-migration/pull/782/files#diff-652fb193d39d942beac21346810bf922R34

**Review by rogyar — CHANGES_REQUESTED** [review]
Hi @ProkopovVitaliy. Great job, thank you. Please, take a look at some required minor adjustments.

**Comment by rogyar** [comment]
Also, please, take a look at the failing static tests

**Comment by rogyar** [comment]
@magento run all tests

**Comment by rogyar** [comment]
Hi @ProkopovVitaliy. There are still issues with the static tests left. Please, take a look

**Comment by ProkopovVitaliy** [comment]
@magento run all tests

File: app/code/Magento/Bundle/view/frontend/templates/catalog/product/view/type/bundle/option/checkbox.phtml [hunk 4]
```diff
@@ -14,16 +14,17 @@
...
                 <input type="hidden"
                        class="bundle-option-<?= $block->escapeHtmlAttr($_option->getId()) ?>  product bundle option"
                        name="bundle_option[<?= $block->escapeHtml($_option->getId()) ?>]"
                        value="<?= $block->escapeHtmlAttr($_selections[0]->getSelectionId()) ?>"/>
-            <?php else :?>
-                <?php foreach ($_selections as $_selection) : ?>
+            <?php else: ?>
+                <?php foreach ($_selections as $_selection): ?>
                     <div class="field choice">
+                        <?php /** phpcs:disable */ ?>
```
Comment (rogyar) [inline app/code/Magento/Bundle/view/frontend/templates/catalog/product/view/type/bundle/option/checkbox.phtml]: Hi @ProkopovVitaliy. Disabling static checks allowed only for some rare cases. We cannot disable the static code checks for fixing static test :)
If there's an issue in the code style it needs to be fixed by adjusting/refactoring the problematic code.

**Review by rogyar — CHANGES_REQUESTED** [review]

File: app/code/Magento/Bundle/view/frontend/templates/catalog/product/view/type/bundle/option/checkbox.phtml (hunk 4 above)
Comment (ProkopovVitaliy) [reply app/code/Magento/Bundle/view/frontend/templates/catalog/product/view/type/bundle/option/checkbox.phtml]: @rogyar  This template really looks scary 😅. 
Will fix it.

**Comment by ProkopovVitaliy** [comment]
@magento run all tests

File: app/code/Magento/Bundle/view/frontend/templates/catalog/product/view/type/bundle/option/checkbox.phtml (hunk 4 above)
Comment (ProkopovVitaliy) [reply app/code/Magento/Bundle/view/frontend/templates/catalog/product/view/type/bundle/option/checkbox.phtml]: @rogyar I refactored template. Also  I think failing tests are not related to the changes presented in the current PR 🤔.

**Comment by rogyar** [comment]
@magento run Functional Tests EE

**Comment by rogyar** [comment]
@magento run all tests

**Comment by rogyar** [comment]
@magento run Unit Tests

**Review by rogyar — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-7655](https://jira.corp.magento.com/browse/ENGCOM-7655) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Manual testing scenario:**

1. Create a bundle product
2. Add option in **Bundle Items** section:

-  Option Name: Test1;
- Input Type: **Checkbox**;
- **Required**
Screenshot from 2020-06-12 09-42-41

3. Add products to Option;
4. Go to Storefront;
5. Open the bundle product page;
6. Click on **Add to Cart** button (no option selected)

**Before:** :heavy_multiplication_x:  Show many messages

Screenshot from 2020-06-12 09-44-31

**After:** :heavy_check_mark:  JS checked and showed one error message for Option

Screenshot from 2020-06-12 09-54-30

Checked also and on Blank theme :heavy_check_mark: 

Screenshot from 2020-06-12 10-20-53

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by jmcharadva** [comment]
@ProkopovVitaliy 
Still there is one issue in this solution. Its working when you are using single group of checkbox section but it's not working while you have multiple sections. Please check it.

**Pull request merged**